cellrank/tl/_lineage.py

```diff
@@ -706,6 +706,44 @@ def copy(self, _="C") -> "Lineage":
     def __copy__(self):
         return self.copy()
 
+    def priming_degree(self, early_cells: Optional[np.ndarray] = None) -> np.ndarray:
+        """
+        TODO.
+
+        Parameters
+        ----------
+        early_cells
+            Indices or a mask marking early cells. If `None`, use all cells.
```

Comment:
I don't agree with this - I think we need a way to specify the set of early cells on the cluster level. I know this makes it difficult to implement this into the Lineage class, which is why I think we should integrate it directly into the base estimator. Using clusters is the natural way to specify a set of early cells.

Reply:
I'd implement it in both, since this really a property of a lineage and delegate.
I wanted to avoid inconsistencies between the 2, though it would be just as simple to do `early_cells=adata.obs['clusters']`.isin(['foo', 'bar])). However, there's also a different problem:
- 1. we'd either have to supply `cluster_key: str` to know from which clustering to take the clusters
- 2. or we'd have to save `cluster_key` when calling `compute_macrostates`, though this is used only for naming and I would advise against it (can also be `None`)
- 3. combination of both (also agains this)

I really don't want to introduce additional argument just for the cluster key, how about a compromise?:
- you can ALSO pass `{'cluster_key': ['foo', 'bar']}` and it will select only cells in `adata.obs['cluster_key'].isin(['foo', 'bar'])`